lib/containers/common.pm

```diff
@@ -90,45 +90,56 @@ sub install_podman_when_needed {
     systemctl("stop firewalld") if ($host_os =~ 'tumbleweed');
 }
 
-sub install_docker_when_needed {
+sub install_docker_multiplatform {
```

Comment:
Not sure why this new function is needed.

Reply:
That's a refactor of the `install_docker_when_needed` function to deal with the perlcritic nesting check failing. Also reduces the `elsif` hell by using early returns.